base/subarray.jl

```diff
@@ -107,7 +107,9 @@ _maybe_reshape_parent(A::AbstractArray, ::NTuple{N, Bool}) where {N} = reshape(A
 Like [`getindex`](@ref), but returns a view into the parent array `A` with the
 given indices instead of making a copy.  Calling [`getindex`](@ref) or
 [`setindex!`](@ref) on the returned `SubArray` computes the
-indices to the parent array on the fly without checking bounds.
+indices to the parent array on the fly without checking bounds. Note that bounds
+are still checked on the indices themselves – `view(x, 10:20)[0]` will throw an
+error because `0` is not a valid index for the range `10:20`."
```

Comment:
Thanks for doing this!

One small correction: it's because `0` is not a valid index in the range `1:11`. For `v = view(x, r)`, `v[i] = x[r[i]]` so you're indexing into `10:20` which has valid indices of `1:11`.

Reply:
Yes, that's what I meant, but you're right that it's a confusing example. Maybe something like...

Note that bounds are still checked on the indices themselves – `view(x, 10:20)[0]` will throw an error since the valid indices for the range `10:20` start at `1`:

```
julia> r = 10:20
10:20

julia> r[1]
10

julia> r[0]
ERROR: BoundsError: attempt to access 11-element UnitRange{Int64} at index [0]
```

(@StefanKarpinski)